packages/core/framework/src/http/types.ts

```diff
@@ -123,12 +123,7 @@ export interface MedusaRequest<Body = unknown>
    */
   filterableFields: Record<string, unknown>
   includes?: Record<string, boolean>
-  /**
-   * An array of fields and relations that are allowed to be queried, this can be set by the
-   * consumer as part of a middleware and it will take precedence over the defaultAllowedFields
-   * @deprecated use `allowed` instead
-   */
-  allowedFields?: string[]
+
```

Comment:
sugg: 

While you're here, you can also remove these from the request:

```
allowedProperties, listConfig, retreiveConfig, validatedQuery, validatedQuery (potentially)
```

Reply:
What are the alternative properties for these?

Reply:
`listConfig, retreiveConfig` => `fields`
`validatedQuery` => `filterableFields`

`allowedProperties` don't know what this is. We don't seem to be using them anywhere

Reply:
it was to configure what can be returned in v1, it was used to clean up the response before sending it back to the customer I think 😂

Reply:
I think, we will have to do a huddle for this, because I cannot find any sibling property named `fields` that has the similar shape as `listConfig`. Maybe I am looking at the wrong place :)

Reply:
Sure, happy to. Also, feel free to move on, this isn't important to address in this PR.

In v1, we used `listConfig` and `retrieveConfig` to access what is `remoteQueryConfig.fields` today. The remoteQueryConfig makes the previous 2 config's redundant.